applications/DEMApplication/python_scripts/DEM_procedures.py

```diff
@@ -1463,6 +1463,8 @@ def AddGlobalVariables(self):
         if self.DEM_parameters["PostParticleMoment"].GetBool():
             self.PushPrintVar(self.PostParticleMoment, PARTICLE_MOMENT, self.global_variables)
         self.PushPrintVar(self.PostDeltaDisplacement, DELTA_DISPLACEMENT, self.global_variables)
+        self.PushPrintVar(self.PostTemperature, TEMPERATURE, self.global_variables)
```

Comment:
Ideally, this should be in the new application. There are ways to do it. You can use your own Thermal_DEM_procedures.py which derives from this one and just adds this modification. If there is a Thermal DEM app, TEMPERATURE should appear nowhere in the DEM app.

Reply:
Same for HEAT_FLUX

Reply:
Yes, in fact a have a subclass for doing exactly what you suggested, but it is not currently adding TEMPERATURE and HEAT_FLUX as these variables are still needed in the DEM Application because of the old thermal particles that were left there. When I remove these particles from the DEM App, I will also bring all the thermal variables.